sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -504,6 +504,19 @@ object LikeSimplification extends Rule[LogicalPlan] {
             Like(input, Literal.create(pattern, StringType))
         }
       }
+
+    case Like(Literal(pattern, StringType), input) =>
+      if (pattern == null) {
+        // If pattern is null, return null value directly, since "null like col" == null.
```

Comment:
This is rather like `If string input is null, return null value directly, ... `.